lib/celluloid/calls.rb

```diff
@@ -23,9 +15,13 @@ def execute_block_on_receiver
     end
 
     def dispatch(obj)
-      check(obj)
-      _b = @block && @block.to_proc
-      obj.public_send(@method, *@arguments, &_b)
+      begin
+        check(obj)
+      rescue => error
+        raise AbortError, error
```

Comment:
From what I can see this is already being tested here: https://github.com/elabs/celluloid/blob/0.17.5-release_allow-receiver-blocks-in-exclusive-mode/spec/celluloid/calls_spec.rb — ArgumentError and NoMethodErrors properly crashing the sender and not the receiver. If I recall correctly (it's been a few months, memory rusty), the AbortError is specifically to crash the sender and not the receiver. Catching the errors when doing check, as opposed to inside the check-method, should make no difference.

Furthermore I tried rebasing this on 0.18.0-prerelease, but I had some conflicts so I stopped that endeavour.

Finally, I'm sorry, but I believe I'll personally be dropping this issue. Thanks for the work you've put into all of this, but at least for the coming few weeks/months I'm done.